source/array_table_model.ts

```diff
@@ -20,11 +20,15 @@ export class ArrayTableModel extends TableModel {
    *  the parent transaction.
    */
   public beginTransaction(): void {
+    this.transactionCount = this.transactionCount + 1;
   }
 
   /** Ends a transaction. */
   public endTransaction(): void {
-    this.dispatcher.dispatch(this.operations);
+    this.transactionCount = this.transactionCount - 1;
+    if(this.transactionCount === 0) {
+      this.dispatcher.dispatch(this.operations);
```

Comment:
This is a bug, create a unit test that performs two transactions in sequence (not recursively) and validate the operations. For example perform an addRow transaction and ensure the operations published is a single ```AddRowOperation```, and then remove that same row and ensure the published operations is a single ```RemoveRowOperation```.

Correct the bug you identify.

Reply:
I'm not seeing what the error is.

Reply:
Push your unit test.